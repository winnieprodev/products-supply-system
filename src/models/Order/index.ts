export * from "./enums";
export * from "./Order.d";
