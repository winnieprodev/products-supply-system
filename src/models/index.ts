export * from "./Cart";
export * from "./Order";
export * from "./Outlet";
export * from "./Product";
export * from "./Search";
export * from "./Staff";
export * from "./Supplier";
export * from "./User";
export * from "./utils";
