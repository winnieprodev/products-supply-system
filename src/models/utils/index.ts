export * from "./async.d";
export * from "./enums";
