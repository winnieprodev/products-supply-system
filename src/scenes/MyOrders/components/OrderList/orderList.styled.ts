import styled from "styled-components";
import { monoTextColor, primaryColor } from "styles/colors";

export const Wrapper = styled.div`
  margin: 30px 0;
`;

export const OrderSection = styled.div`
  padding-left: 20px;
  margin: 30px 0;
  display: flex;
  align-items: center;
`;

export const MonoSection = styled.div`
  display: flex;
  padding: 2px;
  align-items: center;
  text-align: center;
  width: 45px;
  height: 45px;
  background: #f4f4f4;
  border-radius: 10px;
`;

export const MonoText = styled.span`
  margin: auto;
  color: ${monoTextColor};
  font-size: 25px;
  font-weight: bold;
  text-transform: uppercase;
`;

export const CreatedSection = styled.div`
  padding-left: 15px;
  padding-right: 15px;
  width: 35%;
`;

export const SupplierTitle = styled.div`
  color: #0b0b0b;
  font-size: 16px;
  font-weight: 700;
  padding: 5px 0;
`;

export const CreatedText = styled.div`
  font-size: 12px;
  font-weight: 400;
  padding: 5px 0;
`;

export const ItemsSection = styled.div`
  padding-left: 35px;
  padding-right: 15px;
  text-align: right;
  width: 25%;
`;

export const ItemNums = styled.div`
  font-size: 14px;
  font-weight: 600;
  color: ${primaryColor};
  padding: 5px 0;
`;

export const DeliveryText = styled.div`
  font-size: 12px;
  font-weight: 400;
  padding: 5px 0;
`;

export const PrimaryAction = styled.div`
  padding-left: 10px;
  padding-right: 10px;
  width: 150px;
`;

export const SecondaryActions = styled.div`
  padding-left: 10px;
  padding-right: 10px;
  width: 150px;
`;
