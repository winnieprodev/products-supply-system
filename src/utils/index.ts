export * from "./routeUtils";
export * from "./supplierUtils";
export * from "./testUtils";
