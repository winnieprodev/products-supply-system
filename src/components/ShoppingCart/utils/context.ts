import { Async, Cart, ProductCartInfo } from "models";
import React from "react";
import { mockAsync, mockCart } from "utils";

interface ShoppingCartContextProps {
  cartId: Async<number>;
  cart: Async<Cart> | undefined;
  setCart: (cart: Async<Cart> | undefined) => void;
  setProduct: (product: ProductCartInfo) => void;
  removeProduct: (productId: string) => void;
}

export const EMPTY_CART = { products: [] };
const mockFunction = () => {};

export const ShoppingCartContext = React.createContext<ShoppingCartContextProps>(
  {
    cartId: mockAsync(0),
    cart: mockAsync(mockCart(0)),
    setCart: mockFunction,
    setProduct: mockFunction,
    removeProduct: mockFunction,
  },
);
