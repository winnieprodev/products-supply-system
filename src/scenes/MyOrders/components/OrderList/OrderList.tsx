import { Button } from "components";
import { Order, OrderState } from "models";
import React from "react";

import * as Styled from "./orderList.styled";

interface OwnProps {
  items: Order[];
}

const OrderList: React.FC<OwnProps> = ({ items }) => {
  return (
    <Styled.Wrapper>
      {items.map((item) => (
        <Styled.OrderSection key={item.id}>
          <Styled.MonoSection>
            <Styled.MonoText>{item.supplier.mono}</Styled.MonoText>
          </Styled.MonoSection>
          <Styled.CreatedSection>
            <Styled.SupplierTitle>{item.supplier.title}</Styled.SupplierTitle>
            <Styled.CreatedText>
              <b>created:</b> {item.createdDate}
            </Styled.CreatedText>
          </Styled.CreatedSection>
          <Styled.ItemsSection>
            <Styled.ItemNums>{item.itemCount} items</Styled.ItemNums>
            {item.finishedDate && (
              <Styled.DeliveryText>
                <b>
                  {item.state == OrderState.EXPECTED
                    ? item.state + " delivery"
                    : item.state}
                  :
                </b>{" "}
                {item.finishedDate}
              </Styled.DeliveryText>
            )}
          </Styled.ItemsSection>
          <Styled.PrimaryAction>
            <Button primary fullWidth>
              Order again
            </Button>
          </Styled.PrimaryAction>
          <Styled.SecondaryActions>
            {item.state === OrderState.EXPECTED ? (
              <Button primary fullWidth>
                Edit order
              </Button>
            ) : (
              <Button transparent fullWidth>
                Report issue
              </Button>
            )}
          </Styled.SecondaryActions>
        </Styled.OrderSection>
      ))}
    </Styled.Wrapper>
  );
};

export default OrderList;
