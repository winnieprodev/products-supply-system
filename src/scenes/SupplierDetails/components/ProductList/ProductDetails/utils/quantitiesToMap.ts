import { Map } from "immutable";
import { CartQuantity, QuantityType } from "models";

export const quantitiesToMap = (quantities: CartQuantity[]) => {
  return Map<QuantityType, number>(
    quantities.map(({ type, amount }) => [type, amount] as const) as Iterable<
      [QuantityType, number]
    >,
  );
};
