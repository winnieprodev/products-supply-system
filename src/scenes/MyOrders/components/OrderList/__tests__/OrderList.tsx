import { render } from "@testing-library/react";
import { mockOrder } from "utils";
import { OrderState } from "models";

import OrderList from "../OrderList";

describe("OrderList", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot for basic use", () => {
      const { container } = render(
        <OrderList
          items={[
            mockOrder(
              "order1",
              "15/03/2021",
              OrderState.EXPECTED,
              "18/03/2021",
            ),
          ]}
        />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
