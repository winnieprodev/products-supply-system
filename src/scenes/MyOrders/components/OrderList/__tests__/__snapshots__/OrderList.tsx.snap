// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList Snapshot tests it matches the snapshot for basic use 1`] = `
.c11 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #F25C05;
  color: #FFFFFF;
}

.c11:hover {
  background-color: #DD5202;
}

.c11:active {
  background-color: #FF7626;
}

.c12 {
  border: 0;
  border-radius: 8px;
  padding: 8px 16px;
  margin: 4px;
  font-weight: 700;
  width: 100%;
  margin: 0;
}

.c0 {
  margin: 30px 0;
}

.c1 {
  padding-left: 20px;
  margin: 30px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 2px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  width: 45px;
  height: 45px;
  background: #f4f4f4;
  border-radius: 10px;
}

.c3 {
  margin: auto;
  color: rgba(0,0,0,0.25);
  font-size: 25px;
  font-weight: bold;
  text-transform: uppercase;
}

.c4 {
  padding-left: 15px;
  padding-right: 15px;
  width: 35%;
}

.c5 {
  color: #0b0b0b;
  font-size: 16px;
  font-weight: 700;
  padding: 5px 0;
}

.c6 {
  font-size: 12px;
  font-weight: 400;
  padding: 5px 0;
}

.c7 {
  padding-left: 35px;
  padding-right: 15px;
  text-align: right;
  width: 25%;
}

.c8 {
  font-size: 14px;
  font-weight: 600;
  color: #F25C05;
  padding: 5px 0;
}

.c9 {
  font-size: 12px;
  font-weight: 400;
  padding: 5px 0;
}

.c10 {
  padding-left: 10px;
  padding-right: 10px;
  width: 150px;
}

.c13 {
  padding-left: 10px;
  padding-right: 10px;
  width: 150px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        BF
      </span>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        Ben's Food
      </div>
      <div
        class="c6"
      >
        <b>
          created:
        </b>
         
        15/03/2021
      </div>
    </div>
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        2
         items
      </div>
      <div
        class="c9"
      >
        <b>
          Expected delivery
          :
        </b>
         
        18/03/2021
      </div>
    </div>
    <div
      class="c10"
    >
      <button
        class="c11 c12"
      >
        Order again
      </button>
    </div>
    <div
      class="c13"
    >
      <button
        class="c11 c12"
      >
        Edit order
      </button>
    </div>
  </div>
</div>
`;
